ThrowableError.stackTrace() result should include suppressed exceptions

## Is your feature request related to a problem? Please describe.
Java Agent implements its own "print stack trace" that ignores `Throwable#suppressedExceptions` 

## Feature Description
Add suppressed exception to the stack trace captured [here](https://github.com/newrelic/newrelic-java-agent/blob/main/newrelic-agent/src/main/java/com/newrelic/agent/errors/ThrowableError.java#L109)

An example of what `Throwable.prinstStackTrace()` displays for a throwable with suppressed exceptions:
```java
Throwable exception = new RuntimeException("kaboom!", new IllegalArgumentException("kaboom cause"));
exception.addSuppressed(new IllegalStateException("kaboom suppressed 1"));
exception.addSuppressed(new NullPointerException("kaboom suppressed 2"));
exception.printStackTrace()
```
```
java.lang.RuntimeException: kaboom!
	at REPL.$JShell$11.do_it$Aux($JShell$11.java:6)
	at REPL.$JShell$11.do_it$($JShell$11.java:11)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at jdk.jshell/jdk.jshell.execution.DirectExecutionControl.invoke(DirectExecutionControl.java:209)
	at jdk.jshell/jdk.jshell.execution.RemoteExecutionControl.invoke(RemoteExecutionControl.java:116)
	at jdk.jshell/jdk.jshell.execution.DirectExecutionControl.invoke(DirectExecutionControl.java:119)
	at jdk.jshell/jdk.jshell.execution.ExecutionControlForwarder.processCommand(ExecutionControlForwarder.java:144)
	at jdk.jshell/jdk.jshell.execution.ExecutionControlForwarder.commandLoop(ExecutionControlForwarder.java:262)
	at jdk.jshell/jdk.jshell.execution.Util.forwardExecutionControl(Util.java:76)
	at jdk.jshell/jdk.jshell.execution.Util.forwardExecutionControlAndIO(Util.java:137)
	at jdk.jshell/jdk.jshell.execution.RemoteExecutionControl.main(RemoteExecutionControl.java:70)
	Suppressed: java.lang.IllegalStateException: kaboom suppressed 1
		at REPL.$JShell$12.do_it$($JShell$12.java:5)
		... 12 more
	Suppressed: java.lang.NullPointerException: kaboom suppressed 2
		at REPL.$JShell$13.do_it$($JShell$13.java:5)
		... 12 more
Caused by: java.lang.IllegalArgumentException: kaboom cause
	... 14 more
	... 11 more

```
## Describe Alternatives
???

## Additional context
We are using [ZIO](https://zio.dev) which provides fiber traces (cross thread traces) attached to its errors, which are extremely useful in async code. 
For interoperability with other JVM code, ZIO fiber traces are attached to regular throwables.
In the past versions they were added as an extra throwable in the cause chain. This is kind of a hack, and some times requires the use of reflection.
In the latest ZIO release this has been changed in favor of using suppressed exceptions.
This means that once we upgrade ZIO, we will stop seeing fiber traces in NewRelic errors.   

## Priority
Really Want

--- a/newrelic-agent/src/main/java/com/newrelic/agent/errors/ThrowableError.java
+++ b/newrelic-agent/src/main/java/com/newrelic/agent/errors/ThrowableError.java
@@ -18,6 +18,7 @@ import java.util.Collection;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 public class ThrowableError extends TracedError {
 
@@ -112,6 +113,14 @@ public class ThrowableError extends TracedError {
                     stackTrace.add(" caused by " + t.getClass().getName() + ": " + errorMessageReplacer.getMessage(t));
                 }
                 stackTrace.addAll(StackTraces.stackTracesToStrings(t.getStackTrace()));
+                for (Throwable suppressed : t.getSuppressed()) {
+                    stackTrace.add("\tSuppressed: " + suppressed.getClass().getName() + ": " + errorMessageReplacer.getMessage(suppressed));
+                    stackTrace.addAll(
+                            StackTraces.stackTracesToStrings(suppressed.getStackTrace())
+                                .stream().map(line -> "\t\t" + line)
+                                .collect(Collectors.toCollection(ArrayList::new))
+                    );
+                }
                 t = t.equals(t.getCause()) ? null : t.getCause();
                 inner = true;
             }

--- a/newrelic-agent/src/test/java/com/newrelic/agent/errors/TracedErrorTest.java
+++ b/newrelic-agent/src/test/java/com/newrelic/agent/errors/TracedErrorTest.java
@@ -19,6 +19,7 @@ import org.json.simple.JSONObject;
 import org.junit.AfterClass;
 import org.junit.BeforeClass;
 import org.junit.Test;
+import org.mockito.internal.util.collections.Sets;
 
 import java.io.IOException;
 import java.util.Collection;
@@ -333,6 +334,13 @@ public class TracedErrorTest {
         StackTraceElement[] stack_789 = new StackTraceElement[] { stack7, stack8, stack9 };
         top.setStackTrace(stack_789);
 
+        Exception suppressed1 = new IllegalStateException("ISE");
+        StackTraceElement stack10 = new StackTraceElement("Class10", "Method10", "Class10.java", 10);
+        StackTraceElement stack11 = new StackTraceElement("Class11", "Method11", "Class11.java", 11);
+        StackTraceElement stack12 = new StackTraceElement("Class12", "Method12", "Class12.java", 12);
+        suppressed1.setStackTrace(new StackTraceElement[] {stack10, stack11, stack12 });
+        top.addSuppressed(suppressed1);
+
         ErrorCollectorConfig errorCollectorConfig = ServiceFactory.getConfigService().getDefaultAgentConfig()
                 .getErrorCollectorConfig();
 
@@ -342,13 +350,19 @@ public class TracedErrorTest {
 
         Collection<String> stackTrace = error.stackTrace();
         assertNotNull(stackTrace);
-        assertEquals(13, stackTrace.size());
+        assertEquals(17, stackTrace.size());
         Iterator<String> it = stackTrace.iterator();
         assertTrue(it.next().contains("7"));
         assertTrue(it.next().contains("8"));
         assertTrue(it.next().contains("9"));
-        assertEquals("", it.next().trim());
         String next = it.next();
+        assertTrue(next.contains("Suppressed:"));
+        assertTrue(next.contains(suppressed1.toString()));
+        assertTrue(it.next().contains("10"));
+        assertTrue(it.next().contains("11"));
+        assertTrue(it.next().contains("12"));
+        assertEquals("", it.next().trim());
+        next = it.next();
         assertTrue(next.contains("caused by"));
         assertTrue(next.contains(nested1.toString()));
         assertTrue(it.next().contains("4"));
@@ -369,13 +383,18 @@ public class TracedErrorTest {
     }
 
     private String findNextCausedBy(Iterator<String> it) {
+        final String contains = "caused by";
+        return findNextContaining(it, contains);
+    }
+
+    private String findNextContaining(Iterator<String> it, String contains) {
         for (String value = ""; it.hasNext(); value = it.next()) {
-            if (value != null && value.contains("caused by")) {
+            if (value != null && value.contains(contains)) {
                 return value;
             }
         }
 
-        throw new AssertionError("Ran out of elements looking for 'caused by'!");
+        throw new AssertionError("Ran out of elements looking for '"+ contains +"'!");
     }
 
     @Test
@@ -383,6 +402,8 @@ public class TracedErrorTest {
         Exception nested2 = new NullPointerException("Null Pointer Exception should be stripped");
         Exception nested1 = new IOException("IO Exception should be stripped", nested2);
         Exception top = new RuntimeException("RuntimeException should be stripped", nested1);
+        Exception suppressed = new IllegalStateException("IllegalStateException should be stripped");
+        top.addSuppressed(suppressed);
 
         ErrorCollectorConfig errorCollectorConfig = ServiceFactory.getConfigService().getDefaultAgentConfig()
                 .getErrorCollectorConfig();
@@ -396,7 +417,10 @@ public class TracedErrorTest {
         Collection<String> stackTrace = error.stackTrace();
         assertNotNull(stackTrace);
         Iterator<String> it = stackTrace.iterator();
-        String next = findNextCausedBy(it);
+        String next = findNextContaining(it, "Suppressed:");
+        assertTrue(next, next.contains("Suppressed: java.lang.IllegalStateException: Message removed by New Relic 'strip_exception_messages' setting"));
+        assertFalse(next.contains(suppressed.toString()));
+        next = findNextCausedBy(it);
         assertTrue(next, next.contains("caused by java.io.IOException: Message removed by New Relic 'strip_exception_messages' setting"));
         assertFalse(next.contains(nested1.toString()));
         next = findNextCausedBy(it);
@@ -409,20 +433,31 @@ public class TracedErrorTest {
         Exception nested2 = new NullPointerException("Null Pointer Exception should be stripped");
         Exception nested1 = new IOException("IO Exception should not be stripped", nested2);
         Exception top = new RuntimeException("RuntimeException should be stripped", nested1);
+        Exception suppressed1 = new IllegalStateException("IllegalStateException should be stripped");
+        Exception suppressed2 = new IllegalArgumentException("IllegalArgumentException should not be stripped");
+        top.addSuppressed(suppressed1);
+        nested1.addSuppressed(suppressed2);
 
         ErrorCollectorConfig errorCollectorConfig = ServiceFactory.getConfigService().getDefaultAgentConfig()
                 .getErrorCollectorConfig();
 
         TracedError error = ThrowableError
                 .builder(errorCollectorConfig, null, null, top, System.currentTimeMillis())
-                .errorMessageReplacer(new ErrorMessageReplacer(new StripExceptionConfigImpl(true, Collections.singleton("java.io.IOException"))))
+                .errorMessageReplacer(new ErrorMessageReplacer(new StripExceptionConfigImpl(true,
+                        Sets.newSet("java.io.IOException", IllegalArgumentException.class.getName())))
+                )
                 .build();
 
         assertEquals(error.getMessage(), "Message removed by New Relic 'strip_exception_messages' setting");
         Collection<String> stackTrace = error.stackTrace();
         Iterator<String> it = stackTrace.iterator();
-        String next = findNextCausedBy(it);
+        String next = findNextContaining(it, "Suppressed:");
+        assertTrue(next, next.contains("Suppressed: java.lang.IllegalStateException: Message removed by New Relic 'strip_exception_messages' setting"));
+        assertFalse(next.contains(suppressed1.toString()));
+        next = findNextCausedBy(it);
         assertTrue(next, next.contains("caused by java.io.IOException: IO Exception should not be stripped"));
+        next = findNextContaining(it, "Suppressed:");
+        assertTrue(next, next.contains("Suppressed: java.lang.IllegalArgumentException: IllegalArgumentException should not be stripped"));
         next = findNextCausedBy(it);
         assertTrue(next, next.contains("caused by java.lang.NullPointerException: Message removed by New Relic 'strip_exception_messages' setting"));
         assertFalse(next.contains(nested2.toString()));

--- a/newrelic-agent/src/test/java/com/newrelic/agent/errors/TracedErrorTest.java
+++ b/newrelic-agent/src/test/java/com/newrelic/agent/errors/TracedErrorTest.java
@@ -383,8 +383,7 @@ public class TracedErrorTest {
     }
 
     private String findNextCausedBy(Iterator<String> it) {
-        final String contains = "caused by";
-        return findNextContaining(it, contains);
+        return findNextContaining(it, "caused by");
     }
 
     private String findNextContaining(Iterator<String> it, String contains) {

